kerastuner/engine/hyperparameters.py

```diff
@@ -558,44 +558,61 @@ def register(self, name, type, config):
         self.values[full_name] = value
         return value
 
-    def get(self, name):
+    def get(self, name, default=None):
         """Return the current value of this HyperParameter."""
 
-        # Most common case: not attempting to access a conditional parent
-        # or conditional child.
-        full_name = self._get_name(name)
+        # Fast path for most common case: accessing a non-conditional param.
+        full_name = self._get_name(name, include_cond=False)
```

Comment:
I don't quite understand why `include_cond` would be necessary. It makes the implementation somewhat harder to parse. If the parameter is not conditional, the full name lookup should work fine, no?

Reply:
Removed the first check. Looking at this again, the first two checks can be combined. `include_cond=False` in the for loop is used because the user-facing names don't include the conditions (i.e. "model_type=Linear/lr" can just be accessed as `hp.get("lr")` as long as `hp.get("model_type") == 'Linear'`)